### PR TITLE
samples: led_ws2812: adjustable led update delay

### PR DESCRIPTION
**Pull request by jeppenodgaard**
Add a Kconfig option to adjust update frequency of the LED strip.

### CODE REVIEW

**Review by nordicjm — DISMISSED** [review]

File: samples/drivers/led_ws2812/Kconfig [hunk 1]
```diff
@@ -6,4 +6,14 @@
 config SPI
 	default y
 
+menu "Sample Configuration"
```
Comment (kartben) [inline samples/drivers/led_ws2812/Kconfig:9]:
```suggestion
menu "WS2812 Sample Configuration"
```

File: samples/drivers/led_ws2812/Kconfig [hunk 2]
```diff
@@ -6,4 +6,14 @@
...
+
+config LED_UPDATE_DELAY
```
Comment (kartben) [inline samples/drivers/led_ws2812/Kconfig:11]: Probably a nit: not sure there is any specific guideline or best practice but would it be better to namespace it, ex. APP_LED_UPDATE_DELAY or SAMPLE_WS2812_LED_DELAY. Many (but not all) other samples seem to follow this approach.

**Review by kartben — CHANGES_REQUESTED** [review]
Good idea! Since it's a code sample, where we're trying to be extra didactic, could you please add a brief mention of this option and how to use it (ex. show how to pass it directly to west) in the readme too? Thank you!

File: samples/drivers/led_ws2812/src/main.c [hunk 3]
```diff
@@ -21,7 +21,7 @@ LOG_MODULE_REGISTER(main);
 #define STRIP_NODE		DT_ALIAS(led_strip)
 #define STRIP_NUM_PIXELS	DT_PROP(DT_ALIAS(led_strip), chain_length)
 
-#define DELAY_TIME K_MSEC(50)
+#define DELAY_TIME K_MSEC(CONFIG_LED_UPDATE_DELAY)
```
Comment (kartben) [inline samples/drivers/led_ws2812/src/main.c:24]: needs renaming too -- sorry for the extra work for what you thought would be a trivial change 😑

File: samples/drivers/led_ws2812/README.rst [hunk 4]
```diff
@@ -74,6 +74,9 @@ For example devicetree configurations for each compatible, see
 Some boards are already supported out of the box; see the :file:`boards`
 directory for this sample for details.
 
+The sample updates the LED strip periodically. The update frequency can be
+modified by changing the `SAMPLE_LED_UPDATE_DELAY`.
```
Comment (kartben) [inline samples/drivers/led_ws2812/README.rst:78]:
```suggestion
modified by changing the ``SAMPLE_LED_UPDATE_DELAY``.
```

**Review by kartben — CHANGES_REQUESTED** [review]

File: samples/drivers/led_ws2812/Kconfig (hunk 2 above)
Comment (jeppenodgaard) [reply samples/drivers/led_ws2812/Kconfig]: Good point. I did a `grep` and `SAMPLE` prefix seems a bit more common in occurrences and files count (and I personally also prefer this)

File: samples/drivers/led_ws2812/src/main.c (hunk 3 above)
Comment (jeppenodgaard) [reply samples/drivers/led_ws2812/src/main.c]: My bad. Should be fixed now.

File: samples/drivers/led_ws2812/README.rst (hunk 4 above)
Comment (jeppenodgaard) [reply samples/drivers/led_ws2812/README.rst]: Added `:kconfig:option:`, which has been used elsewhere in the file.
What do you use to edit Kconfig files to help with syntax?

Comment (kartben) [reply samples/drivers/led_ws2812/README.rst]: > Added `:kconfig:option:`, which has been used elsewhere in the file. 

Great call :)

> What do you use to edit Kconfig files to help with syntax?

I am assuming you might mean rst files, not Kconfig? 
If the former, I didn't really find a good solution so far https://marketplace.visualstudio.com/items?itemName=lextudio.restructuredtext gets some of the work done but it's pretty buggy (the kind of buggy where sometimes it becomes impossible to use the SPACE key anymore 😅)
I don't touch Kconfigs often so don't really have any editor support there

**Review by kartben — DISMISSED** [review]

File: samples/drivers/led_ws2812/README.rst (hunk 4 above)
Comment (jeppenodgaard) [reply samples/drivers/led_ws2812/README.rst]: Yes I meant rst files :)
Alright I guess I'll stick with `nvim` for now then.

File: samples/drivers/led_ws2812/README.rst [hunk 5]
```diff
@@ -74,6 +74,9 @@ For example devicetree configurations for each compatible, see
 Some boards are already supported out of the box; see the :file:`boards`
 directory for this sample for details.
 
+The sample updates the LED strip periodically. The update frequency can be
+modified by changing the :kconfig:option:`SAMPLE_LED_UPDATE_DELAY`.
```
Comment (nordicjm) [inline samples/drivers/led_ws2812/README.rst:78]: \*CONFIG_

**Review by nordicjm — CHANGES_REQUESTED** [review]

**Review by nordicjm — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Review by simonguinot — APPROVED** [review]
Thanks !

**Pull request merged**